components/src/main/java/dev/arkbuilders/components/filepicker/ArkFilePickerViewModel.kt

```diff
@@ -53,15 +62,16 @@ internal class ArkFilePickerViewModel(
         }
     }
 
-    fun addRoot(root:Path) {
-        viewModelScope.launch {
+    private fun addRoot(root:Path, addRootCallback: () -> Unit) {
+        viewModelScope.launch(Dispatchers.IO) {
             foldersRepo.addRoot(root)
             refreshRootsWithFavs()
+            addRootCallback()
         }
     }
```

Comment:
There is still no need for a callback. Make a function suspend and you will get sequential execution